src/main/kotlin/com/cognifide/gradle/aem/common/instance/InstanceHttpClient.kt

```diff
@@ -15,8 +15,14 @@ open class InstanceHttpClient(private val aem: AemExtension, val instance: Insta
     init {
         baseUrl.set(instance.httpUrl.get())
         escapeUrl.set(true)
-        authorizationPreemptive.set(true)
-        basicCredentials = instance.credentials
+
+        if (instance.serviceCredentials.isPresent) {
+            val token = aem.ims.generateToken(instance.serviceCredentials.asFile.get())
+            bearerToken.set(token)
```

Comment:
no, once again laziness is lost here

`bearerToken.set(instance.serviceCredentials.map { aem.ims.generateToken(it.asFile) })`

Reply:
now you can see why I decided to introduce gradle provider api/properties also in instance. to be able to model  laziness; and now we are able to use it

Reply:
but i told you I removed it, because it's not needed, when we reach this piece of code we need to have the token generated anyway..

Reply:
but still, laziness is always good. it will protect you when some other code will change. and it will avoid unnecessary computation

Reply:
ok will add it back